### PR TITLE
Rework documentation

### LINKED ISSUE #192: [README] Link to contributions is broken
Link: https://godot-kotl.in/contribution/guidelines.html

image

### PR DESCRIPTION
**Pull request by raniejade**
A substantial rework of the documentation, main changes are:

- fixed a lot of formatting issues (please use soft wrapping when writing documentation!)
- simplified some of the sentences
- reworked navigation
- added toggle to switch between light and dark themes

I haven't done any changes to the landing page yet.

Resolves #192

### CODE REVIEW

**Review by chippmann — CHANGES_REQUESTED** [review]
Also not all knowledge base articles are present yet
Really nice work though! Awesome theme. I really like it. Much better than before :-)

**Comment by chippmann** [comment]
> Also not all knowledge base articles are present yet
> Really nice work though! Awesome theme. I really like it. Much better than before :-)

Scrap the knowledge base articles part. They are in a different PR 🙈

**Review by chippmann — APPROVED** [review]
I would merge this after #189

File: docs/src/doc/user-guide/advanced/kotlin-singleton.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+object MySingleton : GodotStatic {
+    var ref: MyReference? = MyReference()
+    var obj: MyObject = MyObject()
+
+    init{
+        registerAsSingleton()
+    }
+
+    override fun collect(){
+        obj.free()
```
Comment (Humberd) [inline docs/src/doc/user-guide/advanced/kotlin-singleton.md:13]: Shouldn't be `queueFree()`? When I did only `free()` godot was not happy about it

Comment (raniejade) [reply docs/src/doc/user-guide/advanced/kotlin-singleton.md:13]: Not sure, @CedNaru wdyt? since you added this.

**Pull request merged**